mmpose/datasets/dataset_wrappers.py

```diff
@@ -1,7 +1,116 @@
 # Copyright (c) OpenMMLab. All rights reserved.
+import numpy as np
+from mmcv.utils import print_log
+from torch.utils.data.dataset import ConcatDataset as _ConcatDataset
+
 from .builder import DATASETS
 
 
+@DATASETS.register_module()
+class ConcatDataset(_ConcatDataset):
+    """A wrapper of concatenated dataset.
+
+    Same as :obj:`torch.utils.data.dataset.ConcatDataset`, but
+    concat the group flag for image aspect ratio.
+
+    Args:
+        datasets (list[:obj:`Dataset`]): A list of datasets.
+        separate_eval (bool): Whether to evaluate the results
+            separately if it is used as validation dataset.
+            Defaults to True.
+    """
+
+    def __init__(self, datasets, separate_eval=True):
+        super(ConcatDataset, self).__init__(datasets)
+        self.separate_eval = separate_eval
+
+    def clip_by_index(self, results, start_idx, end_idx):
+        clip_res = dict()
+        clip_res['preds'] = results[0]['preds'][start_idx:end_idx, :, :]
+        clip_res['boxes'] = results[0]['boxes'][start_idx:end_idx, :]
+        clip_res['image_paths'] = results[0]['image_paths'][start_idx:end_idx]
+        clip_res['bbox_ids'] = results[0]['bbox_ids'][start_idx:end_idx]
```

Comment:
Bottom-up models may not generate predictions with key `bbox_ids`, so a key check may be needed here

Reply:
get